src/shm/sync_util.c

```diff
@@ -0,0 +1,85 @@
+#include "sync_util.h"
+
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/file.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "../shared/types.h"
+#include "../shared/file_util.h"
+
+char * lockpath_from_pass(char* passw, char* base){
+    char* path = malloc(strlen(passw) + strlen(base)+ strlen(".lock"));
+    strcpy(path, base);
+    strcat(path, passw);
+    strcat(path, ".lock");
+    return path;
+}
+
+int lock_sync_file(char *passw, char* base){
+    struct stat st = {0};
+    if (stat(base, &st) == -1) {
+        mkdir(base, 0770);
+    }
+
+    char* path = lockpath_from_pass(passw, base);
```

Comment:
Needs to be freed, see other comment about *alloc and free.

Reply:
yeah, this could be freed inplace, fixed